packages/experience-builder-sdk/src/blocks/VisualEditorContext.tsx

```diff
@@ -187,6 +201,33 @@ export function VisualEditorContextProvider({
           }
           break;
         }
+        case INCOMING_EVENTS.DesignComponentsAdded: {
+          const {
+            tree,
+            designComponent,
+            designComponentDefinition,
+          }: {
+            tree: CompositionTree;
+            designComponent: Entry;
+            designComponentDefinition: ComponentRegistration['definition'];
+          } = payload;
+          if (designComponent) {
```

Comment:
Is there a case where `designComponent` can be undefined? If yes, I would adjust the type of the variables to have type safety. If no, we can remove this guard clause :)

Reply:
I added it just in case someone calls the event and doesn't send the designComponent (for some reason). If that case occurs, then there's no need updating an undefined entity and all other operations there